library/core/src/mem/maybe_uninit.rs

```diff
@@ -456,8 +454,29 @@ impl<T> MaybeUninit<T> {
     /// // x is initialized now:
     /// let s = unsafe { x.assume_init() };
     /// ```
-    #[unstable(feature = "maybe_uninit_extra", issue = "63567")]
-    #[rustc_const_unstable(feature = "maybe_uninit_extra", issue = "63567")]
+    ///
+    /// Usage of this method that avoids an unsafe:
+    ///
+    /// ```rust
+    /// #![feature(maybe_uninit_extra)]
+    /// use core::pin::Pin;
+    /// use core::mem::MaybeUninit;
+    ///
+    /// struct PinArena<T>(Box<[MaybeUninit<T>]>, usize);
+    ///
+    /// impl <T> PinArena<T> {
+    ///     pub fn push(&mut self, val: T) -> Pin<&mut T> {
+    ///         if self.1 >= self.0.len() {
```

Comment:
I would make the example _slightly_ longer and add:

```rust
impl<T> PinArena<T> {
    pub fn capacity(&self) -> usize {
        self.memory.len()
    }
}
```

and then use it here instead of `self.memory.len()`.